sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveDynamicPruningFilters.scala

```diff
@@ -62,8 +63,19 @@ case class PlanAdaptiveDynamicPruningFilters(
           val broadcastValues = SubqueryBroadcastExec(
             name, index, buildKeys, newAdaptivePlan)
           DynamicPruningExpression(InSubqueryExec(value, broadcastValues, exprId))
-        } else {
+        } else if (onlyInBroadcast) {
           DynamicPruningExpression(Literal.TrueLiteral)
+        } else {
+          // we need to apply an aggregate on the buildPlan in order to be column pruned
+          val alias = Alias(buildKeys(index), buildKeys(index).toString)()
+          val aggregate = Aggregate(Seq(alias), Seq(alias), buildPlan)
+
+          val session = adaptivePlan.context.session
+          val planner = adaptivePlan.context.session.sessionState.planner
+          val sparkPlan = QueryExecution.createSparkPlan(session, planner, aggregate)
```

Comment:
shall we call `QueryExecution.prepareExecutedPlan`? For non-AQE DPP, we will plan the DPP subquery in the rule `PlanSubqueries`, which calls `prepareExecutedPlan`

Reply:
Updated.